api/controllers/game/rematch.js

```diff
@@ -1,89 +1,94 @@
 /**
- * Indicate that the player wants to play again
+ * Endpoint to request/decline playing again
+ * If both players accept rematch, creates
+ * a new game, switching who goes first
+ * 
+ * New game will be ranked/casual based on previous match
+ * with name "firstPlayerUsername VS secondPlayerUsername {p0wins}-{p1Wins}-{stalemates}"
  */
 const gameAPI = sails.hooks['customgamehook'];
 module.exports = async function (req, res) {
   try {
-    const { usr: userId, pNum } = req.session;
-    const { gameId, rematch } = req.body;
+    const { usr: userId } = req.session;
+    const { gameId: oldGameId, rematch } = req.body;
 
-    const game = await Game.findOne({ id: gameId }).populate('players');
-    const oldPNum = game.p0 === userId ? 0 : 1;
-    const gameUpdates = { [`p${oldPNum}Rematch`]: rematch };
-
-    await Game.updateOne({ id: game.id }).set(gameUpdates);
-
-    // In case other player updated the game at the same time
-    const updatedGame = await Game.findOne({ id: game.id }).populate('players');
-
-    const currentMatch = await Match.findOne({ id: updatedGame.match });
-    const shouldNewGameBeRanked = currentMatch?.winner ? false : updatedGame.isRanked;
-
-    const { p0: newP1Id, p1: newP0Id } = updatedGame;
+    let game =  await sails.helpers.lockGame(req.session.game);
```

Comment:
We know lock and unlock the requested game to prevent race conditions around which user is the 2nd one to request a rematch (and therefore to start the new game)